contracts/interfaces/IEverlongEvents.sol

```diff
@@ -0,0 +1,35 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.20;
+
+interface IEverlongEvents {
```

Comment:
I can go either way on this (and we don't need to do it now), but I sort of like having all of the different interfaces own their constituent parts. We can't always do this in Hyperdrive for various reasons, but what do you think about moving these event declarations to each of the individual interfaces?

If we find that things are getting shared between them, we could also go the other direction and have separate interfaces for events, errors, and structs.

Reply:
I actually really liked having it the way you're suggesting, but foundry testing made this impractical since you'd need to implement the interface with the test contract (when you're testing against an internally saved address, this is a problemo). 

Since events are the only thing where this is a restriction (errors+structs can be referenced from an interface), I was kinda planning on leaving it as a one-off, wdyt?

Reply:
Ah yeah, good call.

On the subject of errors and structs, I think that makes sense, but it might be weird to use an error from one mixin in another mixin. For example, let's say that we have contracts `EverlongA` and `EverlongB`, and let's say that `IEverlongA` has an error called `Fail`. We might end up using `Fail` in `EverlongB`, in which case we'd end up doing something like:

```
revert IEverlongA.Fail();
```

in `EverlongB`. I don't love this since we'd need the error in both interfaces to get an accurate slice of the full external interface of the "mixin." 

The same basic problem arises with structs. For example, think of how often things like `MarketState` are invoked in hyperdrive. We sidestep the problem there since the interface is consolidated into `IHyperdrive`.